images/linux/scripts/installers/swift.sh

```diff
@@ -9,7 +9,7 @@ source $HELPER_SCRIPTS/install.sh
 
 # Install
 image_label="$(lsb_release -rs)"
-swift_version=$(curl -s -L -N https://swift.org/download|awk -F"[ <]" '/id="swift-/ {print $4; exit}')
+swift_version=$(curl -s "https://api.github.com/repos/apple/swift/releases/latest" | grep -Po '"tag_name": "swift-\K[0-9.]+')
```

Comment:
Is there a point of using github rather than simply pass the `--compressed` use flag to curl?

Reply:
Yes, the point being that Github release syntax is not likely to change while current way of getting the release can break every time swift.org makes a change. This being one of examples of it.

AKA if I would have to choose one of these options on my projects I would go with github release API and forget about it. While if to leave it in current form there is always risk of it breaking.

Which even way you guys go about it i'm fine with it as long as it's fixed on each release tag :)

Reply:
This wouldn't be such a big deal if Microsoft would publish your release images to the public so they can be used in private agent pools in ADO.
Right now we have to hook to your releases and build our own private agents and if it breaks here, it breaks in private projects which then translates to someone having to go through failed build and implement temporary fixes.
I hope you understand.

Reply:
Actually I started discussion about it here : https://github.com/actions/virtual-environments/discussions/4430
I'm not sure if this was discussed already.

Reply:
@JayDoubleu I have nothing against github api in particular, just curious :)

grep is not the best tool to parse json in general, I'd use jq with something like

```
curl -s "https://api.github.com/repos/apple/swift/releases/latest" | jq -r '.tag_name | match("[0-9.]+").string'
```

Reply:
Actually I thought about using JQ but wasn't sure if it's included in the image by default or installed before swift script run.

Reply:
Jq is installed before, so it's worth using it here